mypyc/irbuild/expression.py

```diff
@@ -209,7 +209,8 @@ def transform_call_expr(builder: IRBuilder, expr: CallExpr) -> Value:
         callee = callee.analyzed.expr  # Unwrap type application
 
     if isinstance(callee, MemberExpr):
-        return translate_method_call(builder, expr, callee)
+        return apply_method_specialization(builder, expr, callee) or \
```

Comment:
I am not sure whether it's fine to adding a specialization here, since it's slightly different from the original semantics.

Reply:
I'm pretty sure this change is required for the feature to work.  What can we do to test that it's safe?

Reply:
Then `val = apply_method_specialization(builder, expr, callee, receiver_typ)` inside `translate_method_call` might be redundant?

Reply:
I confirmed that both need to be present in order to work.  This second call is handling the case where `receiver_typ` is present.